controllers/release/release_controller_test.go

```diff
@@ -0,0 +1,144 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	clientsetscheme "k8s.io/client-go/kubernetes/scheme"
+)
+
+var _ = Describe("Release Controller", func() {
+	var (
+		release    *appstudiov1alpha1.Release
+		reconciler *Reconciler
+		scheme     runtime.Scheme
+		req        ctrl.Request
+
+		//strategy *v1alpha1.ReleaseStrategy
+	)
+
+	BeforeEach(func() {
+
+		release = &appstudiov1alpha1.Release{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: testApiVersion,
+				Kind:       "Release",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "myrelease-",
+				Namespace:    testNamespace,
+			},
+			Spec: appstudiov1alpha1.ReleaseSpec{
+				ApplicationSnapshot: "testsnapshot",
+				ReleaseLink:         "testreleaselink",
+			},
+		}
+
+		ctx := context.Background()
+		Expect(cacheClient.Create(ctx, release)).Should(Succeed())
+
+		req = ctrl.Request{
+			NamespacedName: types.NamespacedName{
+				Namespace: release.Namespace,
+				Name:      release.Name,
+			},
+		}
+
+		webhookInstallOptions := &testEnv.WebhookInstallOptions
+
+		k8sManager, _ = ctrl.NewManager(cfg, ctrl.Options{
+			Scheme:             clientsetscheme.Scheme,
+			Host:               webhookInstallOptions.LocalServingHost,
+			Port:               webhookInstallOptions.LocalServingPort,
+			CertDir:            webhookInstallOptions.LocalServingCertDir,
+			MetricsBindAddress: "0", // this disables metrics
+			LeaderElection:     false,
+		})
+		err := (&appstudiov1alpha1.Release{}).SetupWebhookWithManager(k8sManager)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = (&appstudiov1alpha1.ReleaseLink{}).SetupWebhookWithManager(k8sManager)
+		Expect(err).NotTo(HaveOccurred())
+
+		reconciler = NewReleaseReconciler(cacheClient, &logf.Log, &scheme)
+	})
+
+	AfterEach(func() {
+		ctx := context.Background()
+		cacheClient.Delete(ctx, release)
+	})
+
+	It("can create and return a new Reconciler object", func() {
+		Expect(reflect.TypeOf(reconciler)).To(Equal(reflect.TypeOf(&Reconciler{})))
+	})
+
+	It("can ReconcileHandler receive an adapter and return the result for the handling operation", func() {
+		adapter := NewAdapter(release, ctrl.Log, cacheClient, ctx)
+		result, err := reconciler.ReconcileHandler(adapter)
+		Expect(reflect.TypeOf(result)).To(Equal(reflect.TypeOf(reconcile.Result{})))
+		Expect(err).To(BeNil())
+	})
+
+	It("can Reconcile function fail to prepare the adapter and return the result of the reconcile handling operation", func() {
+		cacheClient.Delete(ctx, release)
+		_, err := reconciler.Reconcile(ctx, req)
+		Expect(err).To(BeNil())
+	})
+
+	It("can Reconcile function prepare the adapter and return the result of the reconcile handling operation", func() {
+		result, err := reconciler.Reconcile(ctx, req)
+		Expect(reflect.TypeOf(result)).To(Equal(reflect.TypeOf(reconcile.Result{})))
+		Expect(err).To(BeNil())
+	})
+
+	It("can setup the cache by adding a new index field to search for ReleaseLinks", func() {
+		err := setupCache(k8sManager)
```

Comment:
The cache has a `Get` function. Maybe you can inspect if the expected fields are catched.

Reply:
I need to take a look on this still.